lib/Debugger.ProtocolHandler/RuntimeImpl.cpp

```diff
@@ -315,6 +317,45 @@ namespace JsDebug
         return true;
     }
 
+    JsValueRef JsObjectToString(JsValueRef value, JsValueRef* error)
+    {
+        JsValueRef globalObject;
+        JsGetGlobalObject(&globalObject);
+
+        JsPropertyIdRef propertyId;
+        if (JsGetPropertyIdFromName(L"JSON", &propertyId) != JsNoError)
+        {
+            return nullptr;
+        }
+        JsValueRef jsonObject;
+        if (JsGetProperty(globalObject, propertyId, &jsonObject) != JsNoError)
+        {
+            return nullptr;
+        }
+
+        if (JsGetPropertyIdFromName(L"stringify", &propertyId) != JsNoError)
+        {
+            JsGetAndClearException(error);
```

Comment:
Where does `error` come from? Why is JsGetAndClearException() used in some places and not others?

Reply:
Error is passed as an argument. From ChakraCommmon.h

 /// <summary>
    ///     Returns the exception that caused the runtime of the current context to be in the
    ///     exception state and resets the exception state for that runtime.
    /// </summary>
    /// <remarks>
    ///     <para>
    ///     If the runtime of the current context is not in an exception state, this API will return
    ///     <c>JsErrorInvalidArgument</c>. If the runtime is disabled, this will return an exception
    ///     indicating that the script was terminated, but it will not clear the exception (the
    ///     exception will be cleared if the runtime is re-enabled using
    ///     <c>JsEnableRuntimeExecution</c>).
    ///     </para>
    ///     <para>
    ///     Requires an active script context.
    ///     </para>
    /// </remarks>
    /// <param name="exception">The exception for the runtime of the current context.</param>
    /// <returns>
    ///     The code <c>JsNoError</c> if the operation succeeded, a failure code otherwise.
    /// </returns>
    CHAKRA_API
        JsGetAndClearException(
            _Out_ JsValueRef *exception);

Reply:
OK, it is an out param. So if I understand correctly, if the previous call resulted in an exception, this would clear the exception, and it needs to be used for any call that results in an exception.

But if that's the case that we don't want this JsObjectToString function to return an exception, I would restructure the code so there is a single return and have it always call JsGetAndClearException() before returning the JsValueRef or nullptr.